src/Cli/dotnet/commands/dotnet-workload/WorkloadInfoHelper.cs

```diff
@@ -61,6 +61,8 @@ internal class WorkloadInfoHelper : IWorkloadInfoHelper
                 shouldLog: false);
 
             WorkloadRecordRepo = workloadRecordRepo ?? Installer.GetWorkloadInstallationRecordRepository();
+
+            DotnetPath = dotnetDir ?? (WorkloadFileBasedInstall.IsUserLocal(DotnetPath, _currentSdkFeatureBand.ToString()) ? userProfileDir : DotnetPath);
```

Comment:
I'll throw out that we don't currently use this in any context in which we'd want the real dotnet root folder, but I can make the change anyway to avoid confusion down the line.